src/components/auth/SetupForm.tsx

```diff
@@ -0,0 +1,158 @@
+      }
+      onSubmit={() => {
+        if (!formHasErrors()) {
+          props.setup(email, firstName, preposition, lastName, gender, password, rememberMe);
+        }
+      }}
+    >
+      <Form.Field
+        error={
+        !validator.isEmail(email)
+      }
+      >
+        <Input
+          value={email}
+          placeholder={t('pages.login.email')}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => changeEmail(e.target.value)}
+          ref={inputRef}
+          required={true}
+          type="email"
+        />
+      </Form.Field>
+      <Form.Field
+        error={validator.isEmpty(firstName)}
+      >
+        <Input
+          placeholder={t('entities.user.props.firstName')}
+          value={firstName}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => changeFirstName(e.target.value)}
+          required={true}
+        />
+      </Form.Field>
+      <Form.Field>
+        <Input
+          placeholder={t('entities.user.props.preposition')}
+          value={preposition}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => changePreposition(e.target.value)}
+        />
+      </Form.Field>
+      <Form.Field
+        error={validator.isEmpty(lastName)}
+      >
+        <Input
+          placeholder={t('entities.user.props.lastName')}
+          value={lastName}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => changeLastName(e.target.value)}
+          required={true}
+        />
+      </Form.Field>
+      <Form.Field
+        error={gender === Gender.UNKNOWN}
+      >
+        <Dropdown
+          placeholder={t('entities.user.props.gender.header')}
+          fluid
+          selection
+          options={[
+            { key: 'male', text: t('entities.user.props.gender.male'), value: Gender.MALE },
+            { key: 'female', text: t('entities.user.props.gender.female'), value: Gender.FEMALE },
+            { key: 'other', text: t('entities.user.props.gender.other'), value: Gender.OTHER },
+          ]}
+          onChange={(_e, data) => changeGender(data.value as Gender)}
+          required={true}
+          color='purple'
+        />
+      </Form.Field>
+      <Form.Field
+        name='password-field'
+        id="form-input-password"
+        control={Input}
+        value={password}
+        type="password"
+        placeholder={t('pages.login.password')}
+        onChange={(e: ChangeEvent<HTMLInputElement>) => changePassword(e.target.value)}
+        required={true}
+        error={
+          !validator.isStrongPassword(password)
+        }
+      />
+      <Form.Field>
+        <Checkbox
+          toggle
+          id="form-input-remember-me"
+          checked={rememberMe}
+          onChange={(_e, data) => changeRememberMe(data.checked as boolean)}
+          label={t('pages.login.rememberMe')}
+        />
+      </Form.Field>
+      <Button
+        fluid
+        primary
+        size="large"
+        type="submit"
+      >
+        {t('pages.setup')}
+      </Button>
```

Comment:
I personally find it nice to disable the submit button until all form constraints are met, i.e. no field contains invalid input anymore.